salt/modules/icinga2.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+'''
+Module to provide icinga2 compatibility to salt.
+
+:depends:   - icinga2 server
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+import os
+import subprocess
+
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load this module if the mysql libraries exist
+    '''
+    # TODO: icinga2 check should be os independent and maybe based on package, not binary.
+    if os.path.isfile('/usr/sbin/icinga2'):
+        return True
+    return (False, 'Icinga2 not installed.')
+
+
+def _execute(cmd, ret_code=False):
+    process = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    if ret_code:
+        return process.wait()
+    output, error = process.communicate()
+    if output:
+        log.debug(output)
+        return output
+    log.debug(error)
+    return error
+
+
+def generate_ticket(domain):
+    '''
+    Generate and save an icinga2 ticket.
+
+    Returns::
+        icinga2 pki ticket --cn domain.tld
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' icinga2.generate_ticket domain.tld
+
+    '''
+    result = _execute(["icinga2", "pki", "ticket", "--cn", domain])
+    return result
+
+
+def generate_cert(domain):
```

Comment:
That is the default on debian, ubuntu, centos and fedora. When calling icinga2 in order to generate the certificates, it will ensure the folder is there. Well, at least from one of my tests.
Btw, icinga config has the following defined for all (crt, key, etc): ```cert_path = SysconfDir + "/icinga2/pki/" + NodeName + ".crt"```
And SysconfDir is always /etc if it is installed from packages.
If compiled it will most likely fail unless compiled with /etc as config dir.
So it might be a good addition for users compiling icinga2 but to be honest I excluded this since the automation is a lot more robust while using packages. The state has to install only a few packages and that is all.